backend/btrixcloud/basecrawls.py

```diff
@@ -137,14 +137,40 @@ async def get_resource_resolved_raw_crawl(
         )
         return res
 
+    async def _update_crawl_collections(
+        self, crawl_id: str, org: Organization, collection_ids: List[UUID4]
+    ):
+        """Update crawl collections to match updated list."""
+        crawl = await self.get_crawl(crawl_id, org, cls_type=CrawlOut)
+
+        prior_coll_ids = set(crawl.collectionIds)
+        updated_coll_ids = set(collection_ids)
+
+        # Add new collections
+        added = list(updated_coll_ids.difference(prior_coll_ids))
+        for coll_id in added:
+            await self.colls.add_crawls_to_collection(coll_id, [crawl_id], org)
+
+        # Remove collections crawl no longer belongs to
+        removed = list(prior_coll_ids.difference(updated_coll_ids))
+        for coll_id in removed:
+            await self.colls.remove_crawls_from_collection(coll_id, [crawl_id], org)
+
```

Comment:
one small optimization, not sure if worth it at this time, could be to do `asyncio.gather()` on all of these so they can run in parallel, but may be too early to see if we even need this.

Reply:
Ah nice, `asyncio.gather()` is new to me! I think it's unlikely we'd have more than a few collections here, so this should be pretty quick. Think I'll merge as-is for now but nice to have this note on optimization to come back to if/when we need it!